packages/core/src/fund.ts

```diff
@@ -22,15 +23,25 @@ export const estimateExecutionGas = async (
   ).length
   const estimatedGas = BigInt(150_000) * BigInt(numSetStorageActions)
 
+  const numCallActions = actions.filter((action) => isCallAction(action)).length
+  const estimatedCallActionGas = BigInt(250_000) * BigInt(numCallActions)
+
   const deployedContractPromises = actions
     .filter((action) => isDeployContractAction(action))
     .map(async (action: DeployContractAction) => {
       if (await isContractDeployed(action.addr, provider)) {
         return BigInt(0)
       } else {
-        return provider.estimateGas({
-          data: action.code,
-        })
+        try {
+          // We estimate the gas for the contract deployment by calling `estimateGas` on the provider.
+          return await provider.estimateGas({
+            data: action.code,
+          })
+        } catch (e) {
+          // If the estimate fails, we return a default value of 500k gas which is plenty since the actual
+          // deployment will not happen on chain.
+          return BigInt(500_000)
+        }
```

Comment:
Estimating the gas here throws an error if the constructor reverts, so we catch it here and use a conservative estimate instead. 

Context: We have tooling that attempts to prevent the user from proposing something that will fail during execution. Previously we had similar tooling in the executor as well, where we would check if it seems like a deployment will fail part of the way through and we do not execute it if so. However, since we have an on-chain process for handling if a deployment fails part of the way through, and that process is hit when a post-deployment action fails. It would be better to have consistent behavior where we still attempt the deployment even if a constructor may revert and then run through the same on-chain failure process in all cases.